ui-ngx/src/app/modules/home/components/widget/lib/gateway/connectors-configuration/modbus/modbus-basic-config/modbus-basic-config.component.ts

```diff
@@ -77,13 +81,21 @@ export class ModbusBasicConfigComponent implements ControlValueAccessor, Validat
       master: [],
       slave: [],
     });
+    this.enableSlaveControl = new FormControl(false);
 
     this.basicFormGroup.valueChanges
       .pipe(takeUntil(this.destroy$))
       .subscribe(value => {
         this.onChange(value);
         this.onTouched();
       });
+
+    this.enableSlaveControl.valueChanges
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(enable => {
+        this.updateSlaveEnabling(enable);
+        this.basicFormGroup.get('slave').updateValueAndValidity({emitEvent: !!this.onChange});
```

Comment:
This method is not required when using enabled/disabled form controls

Reply:
it's needed to recheck validity on enable button change